### PR TITLE
Mc 98 100 recorded future new actions

### PR DESCRIPTION
**Pull request by r7-kszczepanskagorna**
### Description

Describe the proposed changes:

  -

### Testing

Any testing information goes here.

## PR Requirements

Developers, verify you have completed the following items by checking them off:

### Style

Review the [style guide](https://komand.github.io/python/style.html)

- [ ] For dependencies, pin [OS package](https://komand.github.io/python/style.html#dockerfile) and [Python package](https://komand.github.io/python/style.html#requirements-txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://komand.github.io/python/sdk.html#sdk-versions) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [ ] For error handling, use of [PluginException](https://komand.github.io/python/error_handling.html#plugin-exceptions) and [ConnectionTestException](https://komand.github.io/python/error_handling.html#connection-exceptions)
- [ ] For logging, use [self.logger](https://komand.github.io/python/sdk.html#logging)
- [ ] For docs, use [changelog style](https://komand.github.io/python/style.html#changelog)
- [ ] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://komand.github.io/python/style.html#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Assessment

<details>

```
[*] Use ``make menu`` for available targets
[*] Including available Makefiles: ../tools/Makefiles/Helpers.mk ../tools/Makefiles/Colors.mk
--
[*] Running validators
[*] Validating plugin at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator LoggingValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Executing validator HelpInputOutputValidator
[*] Executing validator SupportValidator
[*] Executing validator RuntimeValidator
[*] Plugin successfully validated!

----
[*] Total time elapsed: 2629.9139999999998ms

[*] Validating spec with js-yaml
[SUCCESS] Passes js-yaml spec check


[*] Validating markdown...
[SUCCESS] Passes markdown linting

[*] Validating python files for style...
./komand_recorded_future/actions/list_domain_risk_rules/action.py:17:13: F841 local variable 'risklist' is assigned to but never used
./komand_recorded_future/actions/list_hash_risk_rules/action.py:17:13: F841 local variable 'risklist' is assigned to but never used
./komand_recorded_future/actions/list_IP_addresses_risk_rules/action.py:17:13: F841 local variable 'risklist' is assigned to but never used
./komand_recorded_future/actions/list_vulnerability_risk_rules/action.py:17:13: F841 local variable 'risklist' is assigned to but never used
./komand_recorded_future/actions/lookup_IP_address/action.py:47:21: E117 over-indented
./komand_recorded_future/actions/lookup_IP_address/action.py:60:46: F541 f-string is missing placeholders
./komand_recorded_future/actions/lookup_vulnerability/action.py:4:1: E302 expected 2 blank lines, found 1
./komand_recorded_future/actions/lookup_vulnerability/action.py:18:23: E251 unexpected spaces around keyword / parameter equals
./komand_recorded_future/actions/lookup_vulnerability/action.py:18:25: E251 unexpected spaces around keyword / parameter equals
./unit_test/test_lookup_vulnerability.py:5:1: E402 module level import not at top of file
./unit_test/test_lookup_vulnerability.py:6:1: E402 module level import not at top of file
./unit_test/test_lookup_vulnerability.py:7:1: E402 module level import not at top of file
./unit_test/test_lookup_vulnerability.py:8:1: E402 module level import not at top of file
./unit_test/test_lookup_vulnerability.py:9:1: E402 module level import not at top of file
./unit_test/test_lookup_vulnerability.py:27:9: F841 local variable 'e' is assigned to but never used
./unit_test/test_lookup_vulnerability.py:30:1: W293 blank line contains whitespace
./unit_test/test_lookup_vulnerability.py:31:110: W291 trailing whitespace
./unit_test/test_lookup_vulnerability.py:37:9: E303 too many blank lines (2)
./unit_test/test_search_vulnerabilities.py:5:1: E402 module level import not at top of file
./unit_test/test_search_vulnerabilities.py:6:1: E402 module level import not at top of file
./unit_test/test_search_vulnerabilities.py:7:1: E402 module level import not at top of file
./unit_test/test_search_vulnerabilities.py:8:1: E402 module level import not at top of file
./unit_test/test_search_vulnerabilities.py:9:1: E402 module level import not at top of file
./unit_test/test_search_vulnerabilities.py:27:9: F841 local variable 'e' is assigned to but never used
./unit_test/test_search_vulnerabilities.py:30:1: W293 blank line contains whitespace
./unit_test/test_search_vulnerabilities.py:31:110: W291 trailing whitespace
./unit_test/test_search_vulnerabilities.py:37:9: E303 too many blank lines (2)
[FAIL] Fails flake8 linting

[*] Validating python files for security vulnerabilities...
[main]  INFO    profile include tests: None
[main]  INFO    profile exclude tests: None
[main]  INFO    cli include tests: None
[main]  INFO    cli exclude tests: None
[main]  INFO    running on Python 3.7.3
84 [0.. 50.. ]
Run started:2020-09-29 18:25:12.692838

Test results:
        No issues identified.

Code scanned:
        Total lines of code: 11816
        Total lines skipped (#nosec): 0

Run metrics:
        Total issues (by severity):
                Undefined: 0.0
                Low: 0.0
                Medium: 0.0
                High: 0.0
        Total issues (by confidence):
                Undefined: 0.0
                Low: 0.0
                Medium: 0.0
                High: 0.0
Files skipped (0):
[SUCCESS] Passes bandit security checks

[*] Checking for typos with Misspell...
[SUCCESS] Passes Misspell check
```

<summary>
make validate
</summary>
</details>

<details>

```
[*] Validating plugin with all validators at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator LoggingValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Executing validator HelpInputOutputValidator
[*] Executing validator SupportValidator
[*] Executing validator RuntimeValidator
[*] Executing validator ExceptionValidator
[*] Executing validator CredentialsValidator
[*] Executing validator PasswordValidator
[*] Executing validator PrintValidator
[*] Executing validator ConfidentialValidator
[*] Executing validator DockerValidator
[*] Executing validator URLValidator
WARNING: URLs found that return a 4xx code. Verify they are publicly accessible and if not, update with a working URL.
violation: help.md[1258]: https://app.recordedfuture.com/live/sc/entity/ip%!...
violation: help.md[1303]: https://app.recordedfuture.com/live/sc/entity/ip%!...
violation: help.md[86]: http://stix.mitre.org/stix-1
violation: help.md[1211]: http://stix.mitre.org/stix-1
violation: help.md[1467]: http://stix.mitre.org/stix-1
violation: help.md[80]: http://stix.recordedfuture.com/
violation: help.md[1363]: http://stix.recordedfuture.com/
violation: help.md[1464]: http://stix.recordedfuture.com/
violation: help.md[89]: http://stix.mitre.org/TTP-1
violation: help.md[1212]: http://stix.mitre.org/TTP-1
violation: help.md[135]: https://app.recordedfuture.com/live/sc/entity/url%!A(MISSING)http%!A(MISSING)%!F(MISSING)%!F(MISSING)bolizarsospos.com%!F(MISSING)raph9xccgxt
violation: help.md[85]: http://stix.mitre.org/Indicator-2
violation: help.md[1210]: http://stix.mitre.org/Indicator-2
violation: help.md[87]: http://stix.mitre.org/common-1
violation: help.md[1219]: http://stix.mitre.org/common-1
violation: help.md[1468]: http://stix.mitre.org/common-1
violation: help.md[88]: http://stix.mitre.org/default_vocabularies-1
violation: help.md[1366]: http://stix.mitre.org/default_vocabularies-1
violation: help.md[1466]: http://stix.mitre.org/ExploitTarget-1
violation: help.md[1701]: https://api.recordedfuture.com/v2
[*] Plugin successfully validated!

----
[*] Total time elapsed: 89863.946ms
```

<summary>
icon-validate --all .
</summary>
</details>

[assessment.md.txt](https://github.com/rapid7/insightconnect-plugins/files/5301324/assessment.md.txt)

### CODE REVIEW

File: recorded_future/plugin.spec.yaml [hunk 1]
```diff
@@ -1281,3 +1281,140 @@ actions:
...
+        example: Historically Reported by Insikt Group
+    output:
+      data:
+        title: Data
+        description: Search result
+        type: search_data
+        required: true
+  download_url_risk_list:
+    title: Download URL Risk List
+    description: This action returns a risk list of urls matching a filtration
```
Comment (jschipp-r7) [inline recorded_future/plugin.spec.yaml:1377]:
```suggestion
    description: This action returns a risk list of URLs matching a filtration
```

File: recorded_future/plugin.spec.yaml [hunk 2]
```diff
@@ -1281,3 +1281,140 @@ actions:
...
+          - asc
+          - desc
+        required: false
+        example: asc
+      riskScore:
+        title: Risk Score
+        description: Risk score of data
+        type: string
+        required: false
+        example: [0,100]
```
Comment (jschipp-r7) [inline recorded_future/plugin.spec.yaml:1332]: This is an array example, input is a string

**Review by jschipp-r7 — CHANGES_REQUESTED** [review]
